Update: Core versions, v1.1 migration guide, adapter guide (#1417)
fishtown-analytics/docs.getdbt.com: website/docs/docs/contributing/building-a-new-adapter.md

- Hosting on a public git provider (e.g. GitHub, GitLab)
- Publishing to [PyPi](https://pypi.org/)
- Adding to the list of ["Available Adapters"](available-adapters#community-supported)

### Maintaining your new adapter

> If your adapter catches on, and people start using it, you may find yourself as the maintainer of an increasingly popular open source project. It's best to communicate your intentions early and often: Is this experimental work that people should use at their own risk? Or is this production-grade code that you're committed to maintaining into the future?

Reviewer: ```suggestion
When your adapter becomes more popular, and people start using it, you may quickly become the maintainer of an increasingly popular open source project. With this new role, comes some unexpected responsibilities that not only include code maintenance, but also working with a community of users and contributors. To help people understand what to expect of your project, you should communicate your intentions early and often in your adapter documentation or README. Answer questions like, Is this experimental work that people should use at their own risk? Or is this production-grade code that you're committed to maintaining into the future?
```

> Be aware that new minor version releases of `dbt-core` may include changes to the Python interface for adapter plugins, as well as new or updated test cases. The maintainers of `dbt-core` will clearly communicate these changes in documentation and release notes, and they will aim for backwards compatibility whenever possible. Patch releases of `dbt-core` will _not_ include breaking changes to adapter-facing code. (For more details, see ["About dbt Core versions"](core-versions).)

Reviewer: ```suggestion
#### Keeping the code compatible with dbt Core

New minor version releases of `dbt-core` may include changes to the Python interface for adapter plugins, as well as new or updated test cases. The maintainers of `dbt-core` will clearly communicate these changes in documentation and release notes, and they will aim for backwards compatibility whenever possible. 

Patch releases of `dbt-core` will _not_ include breaking changes to adapter-facing code. For more details, see ["About dbt Core versions"](core-versions).
```

> We strongly encourage you to adopt the following approach when versioning and releasing your plugin:

Reviewer: ```suggestion
#### Versioning and releasing your adapter

We strongly encourage you to adopt the following approach when versioning and releasing your plugin:
```
